src/Shell/Task/ModelTask.php

```diff
@@ -781,6 +793,34 @@ public function fieldValidation($schema, $fieldName, array $metaData, $primaryKe
         return $validation;
     }
 
+    /**
+     * Get the type specific allowempty/notempty method suffix.
+     *
+     * @param string $fieldName Field name.
+     * @param array $metaData Field meta data.
+     * @return string
+     */
+    protected function getEmptyMethodType($fieldName, $metaData)
+    {
+        switch ($metaData['type']) {
+            case 'date':
+                return 'Date';
+
+            case 'time':
+                return 'Time';
+
+            case 'datetime':
+            case 'timestamp':
+                return 'DateTime';
+        }
+
+        if (preg_match('/file|image/', $fieldName)) {
+            return 'File';
+        }
+
+        return 'String';
```

Comment:
The alternative would be to use normal "camelBacked" (string, dateTime) and ufcirst() the result as suffix.
This way the "type" return here is true. But no big deal either way.